lib/lang/greek/greek-language-dataset.js

```diff
@@ -436,10 +510,17 @@ export default class GreekLanguageDataset extends LanguageDataset {
     return this.pronounGroupingLemmas.has(grammarClass) ? this.pronounGroupingLemmas.get(grammarClass) : []
   }
 
+  getNumeralGroupingLemmas () {
+    return this.numeralGroupingLemmas
+  }
+
   static getObligatoryMatchList (inflection) {
     if (inflection.hasFeatureValue(Feature.types.part, Constants.POFS_PRONOUN)) {
       // If it is a pronoun, it must match a grammatical class
       return [Feature.types.grmClass]
+    } else if (inflection.hasFeatureValue(Feature.types.part, Constants.POFS_NUMERAL)) {
+      // If it is a numeral, it must match a part of speach
+      return [Feature.types.part]
```

Comment:
Do we also need to add Feature.types.fullForm to the optionalMatchList for numerals in order to ever get an exact match on the form?

Reply:
it catches exact match now without additional adding to optinalMatchList.
(see the upper screenshot).
Or what do you mean?

Reply:
an exact match would be highlighted in yellow, a partial match is blue. An exact match is when all of the features in the obligatoryMatchList are matched AND all of the features in the optionalMatchList are met. I'm not positive though that my analysis is correct about needing to put the form in the optionalMatchList, because if it was, I would have expected both forms to show up matched and not just the one that actually does match. So somewhere form matching is happening, I'm just not sure where.

Reply:
Oh, thank you for detailed explanation! ok, I will check it tomorrow and add to optionalMatch (if needed).

Reply:
sounds good!

Reply:
The full match is determined here: https://github.com/alpheios-project/inflection-tables/blob/master/lib/language-dataset.js#L317-L341

In order to find a full match first thing we check is `matchData.suffixMatch = inflection.compareWithWord(item.value)`. That takes care of matching a full form and thus we don't need to have fullFrom in optionalMatches.

We use a special function here because it takes care of normalization and some other non-linear matching logic.

Reply:
ah, okay, thanks that explains how the form is getting matched then. I think we may still have an issue getting an exact match here, but it is not related to the form then. I would suggest going ahead and merging the PR and then we can look at the reasons for not getting an exact match separately.

Reply:
I have investigated the problem with fullMatch and find out that this problem is only for combined feature's values.
For example if an item has gender = 'masculine femine neuter' - it won't ever have full match, because it compares with features determined in greek-language-dataset.js / getOptionalMatchList. And they have a standard feature's values and, for example, gender doesn't have combined variants.
I have fixed this method for numerals and for pronouns and now it works well:

for numerals:
![image](https://user-images.githubusercontent.com/12377640/42013791-4df96b10-7ae2-11e8-94d7-c9f85d1b5558.png)
![image](https://user-images.githubusercontent.com/12377640/42013828-69f9df16-7ae2-11e8-85aa-069c9a626f54.png)

for pronoun
![image](https://user-images.githubusercontent.com/12377640/42013849-82704148-7ae2-11e8-94db-07ca09472e37.png)